pkg/target/match_schema.go

```diff
@@ -1,130 +1,44 @@
-						Description: "The Group and Kind of objects that should be matched.  If multiple groups/kinds combinations are specified, an incoming resource need only match one to be in scope.",
-						Properties: map[string]apiextensions.JSONSchemaProps{
-							"apiGroups": nullableStringList,
-							"kinds":     nullableStringList,
-						},
-					},
-				},
-			},
-			"namespaces":         *propsWithDescription(&wildcardNSList, "`namespaces` is a list of namespace names. If defined, a constraint only applies to resources in a listed namespace.  Namespaces also supports a prefix-based glob.  For example, `namespaces: [kube-*]` matches both `kube-system` and `kube-public`."),
-			"excludedNamespaces": *propsWithDescription(&wildcardNSList, "`excludedNamespaces` is a list of namespace names. If defined, a constraint only applies to resources not in a listed namespace. ExcludedNamespaces also supports a prefix-based glob.  For example, `excludedNamespaces: [kube-*]` matches both `kube-system` and `kube-public`."),
-			"labelSelector":      *propsWithDescription(&labelSelectorSchema, "`labelSelector` is the combination of two optional fields: `matchLabels` and `matchExpressions`.  These two fields provide different methods of selecting or excluding k8s objects based on the label keys and values included in object metadata.  All selection expressions from both sections are ANDed to determine if an object meets the cumulative requirements of the selector."),
-			"namespaceSelector":  *propsWithDescription(&labelSelectorSchema, "`namespaceSelector` is a label selector against an object's containing namespace or the object itself, if the object is a namespace."),
-			"scope": {
-				Type:        "string",
-				Description: "`scope` determines if cluster-scoped and/or namespaced-scoped resources are matched.  Accepts `*`, `Cluster`, or `Namespaced`. (defaults to `*`)",
-				Enum: []apiextensions.JSON{
-					"*",
-					"Cluster",
-					"Namespaced",
-				},
-			},
-			"name": {
-				Type:        "string",
-				Description: "`name` is the name of an object.  If defined, it matches against objects with the specified name.  Name also supports a prefix-based glob.  For example, `name: pod-*` matches both `pod-a` and `pod-b`.",
-				Pattern:     wildcardNSPattern,
-			},
-		},
+	// Convert v1 JSONSchemaProps to versionless
+	rt := runtime.NewScheme()
+	if err := apiextensions.AddToScheme(rt); err != nil {
+		panic(fmt.Errorf("could not add apiextensions to scheme: %w", err))
+	}
+	if err := apiextensionsv1.AddToScheme(rt); err != nil {
+		panic(fmt.Errorf("could not add apiextensionsv1 to scheme: %w", err))
 	}
+	embedded := matchCRD.Spec.Versions[0].Schema.OpenAPIV3Schema.Properties["embeddedMatch"]
+	if err := rt.Convert(&embedded, &matchJSONSchemaProps, nil); err != nil {
+		panic(fmt.Errorf("could not convert match JSONSchemaProps from v1 to versionless: %w", err))
+	}
+}
+
+func matchSchema() apiextensions.JSONSchemaProps {
+	return matchJSONSchemaProps
```

Comment:
We should return a deepcopy to guard against callers modifying the contents.

Reply:
Has this been addressed?

Reply:
Same as response above: latest code should have all comments addressed.